helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/ClusterAccessor.java

```diff
@@ -888,6 +885,91 @@ public Response updateCloudConfig(@PathParam("clusterId") String clusterId,
+    ClusterConfig config;
+    try {
+      config = getClusterConfigWithConfigAccessor(clusterId);
+    } catch (Exception ex) {
+      LOG.error("Failed to get cluster config for cluster {}. Exception: {}", clusterId, ex);
+      return serverError(ex);
+    }
+    if (config == null) {
+      return notFound();
+    }
+    int threadPoolSize = config.getGlobalTargetTaskThreadPoolSize();
+    if (threadPoolSize == TaskConstants.TARGET_TASK_THREAD_POOL_SIZE_NOT_SET) {
+      return notFound();
+    }
+    Map<String, Integer> resultMap = new HashMap<>();
+    resultMap.put(ClusterProperties.globalTargetTaskThreadPoolSize.name(), threadPoolSize);
+    return JSONRepresentation(resultMap);
+  }
+
+  /**
+   * Update the global target task thread pool size of the cluster. The global target task thread
+   * pool size goes to ClusterConfig, and is applied to all instances of the cluster for task thread
+   * pool construction. If an instance doesn't specify its target thread pool size in
+   * InstanceConfig, then this value in ClusterConfig will be used to construct its task thread
+   * pool. The newly-set target task thread pool size will take effect upon a JVM restart. If none
+   * of the global and per-instance target thread pool sizes are set, a default value -
+   * TaskConstants.DEFAULT_TASK_THREAD_POOL_SIZE will be used.
+   * @param clusterId - the cluster id of which the global target thread pool size is set to
+   * @param threadPoolSize - the thread pool size to be set
+   */
+  @POST
+  @Path("{clusterId}/global-target-task-thread-pool-size/{threadPoolSize}")
+  public Response updateGlobalTargetTaskThreadPoolSize(@PathParam("clusterId") String clusterId,
+      @PathParam("threadPoolSize") int threadPoolSize) {
+    ConfigAccessor accessor = getConfigAccessor();
+    ClusterConfig config;
+    try {
+      config = getClusterConfigWithConfigAccessor(clusterId);
+    } catch (Exception ex) {
+      LOG.error("Failed to get cluster config for cluster {}. Exception: {}", clusterId, ex);
+      return serverError(ex);
+    }
+    if (config == null) {
+      return notFound();
+    }
+    try {
+      config.setGlobalTargetTaskThreadPoolSize(threadPoolSize);
+    } catch (IllegalArgumentException ex) {
+      return badRequest(ex.getMessage());
+    }
+    try {
+      accessor.updateClusterConfig(clusterId, config);
+    } catch (Exception ex) {
+      LOG.error("Failed to update cluster config for cluster {}. Exception: {}", clusterId, ex);
+      return serverError(ex);
+    }
+
+    return OK();
+  }
+
+  /*
+   * Get ClusterConfig by using config accessor. Logs HelixException.
+   */
+  private ClusterConfig getClusterConfigWithConfigAccessor(String clusterId) {
+    ConfigAccessor accessor = getConfigAccessor();
+    try {
+      return accessor.getClusterConfig(clusterId);
+    } catch (HelixException ex) {
+      LOG.info("Failed to get cluster config for cluster {}.", clusterId, ex);
+    }
+
+    return null;
```

Comment:
I'm not sure returning null is a good idea here. Also, I wouldn't refactor this into a private method because it doesn't seem so necessary?

Reply:
Returning null is the only way to refactor here. But in terms of why refactoring, it does get rid of several lines of duplicate code that's in 3 functions - I'm pretty sure if I get rid of this function, someone else will complain about code duplication.